profile/middle_weight_benchmark.rb

```diff
@@ -0,0 +1,233 @@
+$LOAD_PATH.unshift(File.dirname(__FILE__))
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__), "..", "lib"))
+
+require 'mongo'
+require 'benchmark'
+require_relative 'benchmark_helper'
+
+TWITTER = "TWITTER"
+SMALL_DOC = "SMALL_DOC"
+LARGE_DOC = "LARGE_DOC"
+GRIDFS_LARGE = "GRIDFS_LARGE"
+GRIDFSTEST = "gridfstest"
+
+# Perform 'middleweight' benchmarks. This includes
+#
+#   Find many and empty the cursor
+#   Small doc bulk insert
+#   Large doc bulk insert
+#   GridFS upload
+#   GridFS download
+#
+# The featherweight benchmark is intended to measure multi-document insertion and query tasks,
+# to explore batch-write and cursor chunking efficiency.
+#
+# @param [ Integer ] benchmark_reps Number of repetitions of each benchmark to run.
+#
+# @return [ Array< Array<Integer> > ] Arrays of benchmark results
+#
+# @since 2.2.1
+def middleweight_benchmark(benchmark_reps)
+  #bench_helper = BenchmarkHelper.new('perftest','corpus')
+  bench_helper = BenchmarkHelper.new('foo','bar')
+  database = bench_helper.database
+  collection = bench_helper.collection
+
+  results = []
+  results << find_many_and_empty_the_cursor(database, collection, benchmark_reps)
+  results << small_doc_bulk_insert(database, collection, benchmark_reps)
+  #results << large_doc_bulk_insert(database, collection, benchmark_reps)    #TODO: this benchmark takes WAY too long. LARGE_DOC data is too big
+  results << gridfs_upload(database, benchmark_reps)
+  results << gridfs_download(database, benchmark_reps)
+  results
+end
+
+
+# Find many and empty the cursor
+#
+# - Drop the database
+# - Load TWITTER dataset
+# - Insert the first 10,000 documents into the 'corpus' collection.  TODO:QUESTION I changed this to all the Twitter documents. Can't get MB of data otherwise -- need to get that from the data file size, b/c object sizes vary b/w languages/implementations
+#
+# Measure: Issue a find command on the 'corpus' collection with an empty filter expression.
+#          Retrieve (and discard) all documents from the cursor.
+#
+# @param [ Mongo::Database ] database MongoDB Database.
+# @param [ Mongo::Collection ] collection The collection.
+# @param [ Integer ] reps Number of repetitions of the benchmark to run.
+#
+# @return [ [Array<Integer>, Double, String] ] An array of benchmark wall clock time results, the size of the dataset in MB, test label
+#
+# @since 2.2.1
+def find_many_and_empty_the_cursor(database, collection, reps)
+  database.drop
+  twitter_data = BenchmarkHelper.load_array_from_file(TWITTER)
+  data_file_size = File.size(TWITTER)
+  #twitter_data_size = twitter_data.size
+
+  #10000.times do |i|
+  #  collection.insert_one( twitter_data[i] ) if (i < twitter_data_size)
```

Comment:
Actually, the insertion code here isn't part of the timing. That starts with 'Benchmark.bm do...'